robot/server/server.go

```diff
@@ -417,25 +418,38 @@ func (s *Server) Log(ctx context.Context, req *pb.LogRequest) (*pb.LogResponse,
 	}
 	log := req.Logs[0]
 
+	var fields []zap.Field
+	for _, fieldP := range log.Fields {
+		fieldPJSON, err := fieldP.MarshalJSON()
+		if err != nil {
+			return nil, err
+		}
+		var zf zap.Field
+		if err = json.Unmarshal(fieldPJSON, &zf); err != nil {
+			return nil, err
+		}
+		fields = append(fields, zf)
+	}
+
 	// Use a sublogger of robot logger with correct logger name. Set a level of
 	// DEBUG to allow gRPC logs at DEBUG level even when RDK is not on DEBUG
 	// level. Disable caller to mimic caller passed in from gRPC request.
 	logger := s.r.Logger().Sublogger(log.LoggerName)
 	logger.SetLevel(logging.DEBUG)
-	l := logger.WithOptions(zap.WithCaller(false))
+	l := logger.WithOptions(zap.WithCaller(false)).Desugar()
```

Comment:
`Desugar` to allow directly passing `zap.Field`s to `Debug/Warn/Info/Error` calls.